script/tool/lib/src/version_check_command.dart

```diff
@@ -169,12 +178,29 @@ class VersionCheckCommand extends PluginCommand {
     final Iterator<String> iterator = lines.iterator;
     while (iterator.moveNext()) {
       if (iterator.current.trim().isNotEmpty) {
-        firstLineWithText = iterator.current;
+        firstLineWithText = iterator.current.trim();
         break;
       }
     }
     // Remove all leading mark down syntax from the version line.
-    final String versionString = firstLineWithText.split(' ').last;
+    String versionString = firstLineWithText.split(' ').last;
+
+    // Skip validation for the special NEXT version that's used to accumulate
+    // changes that don't warrant publishing on their own.
+    bool hasNextSection = versionString == 'NEXT';
+    if (hasNextSection) {
+      print('Found NEXT; validating next version in the CHANGELOG.');
+      // Ensure that the version in pubspec hasn't changed without updating
+      // CHANGELOG. That means the next version entry in the CHANGELOG pass the
+      // normal validation.
+      while (iterator.moveNext()) {
+        if (iterator.current.trim().startsWith('## ')) {
```

Comment:
some of our CHANGELOG.md files do no follow the standard format, see: https://github.com/flutter/plugins/blob/master/packages/url_launcher/url_launcher_web/CHANGELOG.md

This would fail. :(

Reply:
I'm fine with a model where trying to use NEXT won't work until the format has been fixed to follow our standard format.

We can also do a one-time sweep if that becomes annoying.

Reply:
Sounds good!